src/payment/strategies/sage-pay/sage-pay-payment-strategy.ts

```diff
@@ -22,18 +25,39 @@ export default class SagePayPaymentStrategy extends CreditCardPaymentStrategy {
     }
 
     execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!payload.payment) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        if (this._isThreeDSTwoExperimentOn()) {
+            payload.payment.paymentData = {
+                ...payload.payment.paymentData,
+                browser_info: getBrowserInfo(),
+            };
+        }
+
```

Comment:
```suggestion
        const { payment } = payload;
        if (!payment) {
            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
        }
        
        const { paymentData } = payment;
        
        if (!paymentData) {
            throw new MissingDataError(MissingDataErrorType.MissingPayment);
        }

        if (this._isThreeDSTwoExperimentOn()) {
            paymentData = {
                ....paymentData,
                browser_info: getBrowserInfo(),
            };
        }

```

Reply:
I think this wont work as we are adding parameters to the payload object, we will have to do something like this
 
        const { payment } = payload;
        if (!payment) {
            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
        }

        const { paymentData } = payment;

        if (!paymentData) {
            throw new MissingDataError(MissingDataErrorType.MissingPayment);
        }

        if (this._isThreeDSTwoExperimentOn()) {
            const newPaymentData = {
                ...paymentData,
                browser_info: getBrowserInfo(),
            };

            if ( payload.payment ) {
                payload.payment.paymentData = newPaymentData;
            }
        }
        
and the two ifs will be useless

Reply:
Yes can we do the above then, the ifs are not useless since they act as protection in case the value is null/undefined. Which can happen.

Reply:
The end-use facing rerror messages for `NotInitializedError(NotInitializedErrorType.PaymentNotInitialized)` and `MissingDataError(MissingDataErrorType.MissingPayment)` are different. 

In my opinion, it is a better experience to provide accurate error information.

Besides this, all look good to me.

Reply:
makes senses to me, thanks! changes are addressed